tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -790,6 +845,19 @@ Status FragmentConsolidator::set_query_buffers(
     ++bid;
   }
 
+  if (config_.with_delete_meta_ && !dense) {
```

Comment:
Could you expand a little on your comment? Maybe you have a suggestion on how to make it better? I want to file a SC item but not sure what to put into it.

Reply:
By unwieldy I mean this kind of thing (`bid+1`, `bid += 2` etc.) - I understand what it's doing, but this could be made much clearer and shorter: https://github.com/TileDB-Inc/TileDB/blob/8cf4ea1cfdbac995270e1c152eba77f06f96ef91/tiledb/sm/consolidator/fragment_consolidator.cc#L792-L794

Anyway, let's discuss. There are many existing uses of this "one, maybe two, maybe three"  buffer pattern in various other places internally (eg `QueryBuffer`).